modules/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+class Jetpack_Simple_Payments {
+	// These have to be under 20 chars because that is CPT limit.
+	static $post_type_order = 'jp_pay_order';
+	static $post_type_product = 'jp_pay_product';
+
+	static $shortcode = 'simple-payment';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts() {
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js' );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ) , array( 'paypal-checkout-js' ) );
+	}
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+	private function register_shortcode() {
+		add_shortcode( static::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		$this->register_scripts();
+		$this->register_shortcode();
+		$this->setup_cpts();
+	}
+
+	function parse_shortcode( $attrs, $content = false ) {
+		if( empty( $attrs[ 'id' ] ) ) {
+			return;
+		}
+		$product = get_post( $attrs[ 'id' ] );
+		if( is_wp_error( $product ) ) {
+			return;
+		}
+		if( $product->post_type !== self::$post_type_product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'dom_id' => uniqid( 'jp_simple_payments__button_' . $product->ID . '_' ),
+			'class' => 'jp_simple_payments__' . $product->ID,
+			'title' => get_the_title( $product ),
+			'description' => $product->post_content,
+			'cta' => get_post_meta( $product->ID, 'spay_cta', true ),
+		), $attrs );
+		$data[ 'price' ] = $this->format_price(
```

Comment:
Not a big deal but the WP code guidelines say to not put spaces inside array reference brackets if the key is a string (which is annoying since they say the opposite for JavaScript). There's a few other instances of this syntax in this file. A code linter should notice them.